src/Type.cpp

```diff
@@ -7,33 +7,19 @@ namespace Halide {
 using std::ostringstream;
 
 /** Return an integer which is the maximum value of this type. */
-int Type::imax() const {
+int64_t Type::imax() const {
     if (is_uint()) {
-        if (bits == 32) {
-            return 0xffffffff;
-        } else if (bits < 32) {
-            return (int) ((1 << bits) - 1);
+        if (bits == 64) {
+            return -1;
```

Comment:
Can you give a bit of insight in how this is used? I would expect that one has to case on the type and then use a C cast appropriately before doing anything with the value. The only other approach I can think of is to have two methods, one signed and one unsigned and then assert if the wrong one is called. I'm not sure that is better, but this sort of thing has caused problems for me in the past so I'm bringing it up.